### PR TITLE
docs: :memo: include descriptions and details for the `path` functions

### PR DESCRIPTION
**Pull request by lwjohnst86**
## Description

These changes add the descriptions and explanations for all the path functions

## Reviewer Focus

This PR needs an in-depth review.

### CODE REVIEW

File: docs/design/python-functions.qmd [hunk 1]
```diff
@@ -253,45 +253,61 @@ follows the Frictionless Data specification.
...
-::: callout-warning
-These functions are still being developed.
-:::
+-   Nearly all the below `path_*()` functions use `path_sprout_root()`
+    internally to get the root path as well as `package_id` and/or
+    `resource_id` to complete the correct path for the specific
+    package/resource.
+-   All of these functions output a path object.
+-   The path's returned by all of these functions are paths that exist,
+    so all include a `verify_is_dir()` or `verify_is_file()` check.
```
Comment (signekb) [inline docs/design/python-functions.qmd:262]: Do we need a `verify_exists()` function? Currently, the `verify_is_dir()` only checks whether the path is a directory or not - not whether it exists?

File: docs/design/python-functions.qmd [hunk 2]
```diff
@@ -253,45 +253,61 @@ follows the Frictionless Data specification.
...
+-   If the wrong `package_id` or `resource_id` is given, an error
+    message will include a list of all the actual `package_id` at the
+    `path_sprout_root()` location or `resource_id` for a specific
```
Comment (signekb) [inline docs/design/python-functions.qmd:265]:
```suggestion
    `path_sprout_root()` location or all actual `resource_id` for a specific
```

**Review by signekb — COMMENTED** [review]
Nice! 🎉  Two comments:

File: docs/design/python-functions.qmd (hunk 1 above)
Comment (martonvago) [reply docs/design/python-functions.qmd:262]: Doesn't it throw if the directory doesn't exist, as well?

**Review by martonvago — CHANGES_REQUESTED** [review]
Very nice 👍 👍

File: docs/design/python-functions.qmd (hunk 1 above)
Comment (martonvago) [reply docs/design/python-functions.qmd:262]: > Do we need a `verify_exists()` function? Currently, the `verify_is_dir()` only checks whether the path is a directory or not - not whether it exists?

Doesn't it throw if the directory doesn't exist, as well?

Comment (signekb) [reply docs/design/python-functions.qmd:262]: omg, you're right ... and I have even made the test for it 😝

**Pull request merged**